yt/yt/server/query_tracker/handler_base.cpp

```diff
@@ -81,15 +81,22 @@ TSharedRef TruncateRowset(TSharedRange<NTableClient::TUnversionedRow>& rows)
 
 void ProcessRowset(TFinishedQueryResultPartial& newRecord, TWireRowset wireSchemaAndSchemafulRowset)
 {
+    YT_LOG_DEBUG("Processing wire rowset");
     // Process schema.
-    auto reader = CreateWireProtocolReader(wireSchemaAndSchemafulRowset.Rowset);
+    YT_LOG_DEBUG("Reading schema");
+    TWireProtocolOptions readerOptions;
+    readerOptions.MaxStringValueLength = MaxRowWeightLimit;
+    readerOptions.MaxAnyValueLength = MaxRowWeightLimit;
```

Comment:
Why do we use MaxRowWeightLimit which is marked as "Only for static tables"
https://github.com/ytsaurus/ytsaurus/blob/ccc0a106bda90562675784da9492e08ec5997c3e/yt/yt/client/table_client/public.h#L107

Reply:
Before we used `Only for dynamic tables` limit.

For query tracker I see it as a reasonable limit (`select * from static_table`). I can't image real use case where longer values would become even longer (like `select row + "a" from static_table`)

I can put there `numeric_limits<i64>::max()`, if you want. But would prefer to get some estimation of what we want to support and what QueryTracker can handle (let's not give him 100GB lines)

Feel free to propose a better constant in code, or a new one

Reply:
Maybe we can add a fallback for such cases (like select row + "a" from static_table). If wire protocol reader fails with something like `Value of type "string" is too long` we can at least shorten the result by nothing instead of an infinite loop

Reply:
1. Whatever limit we use, we should not allow the same behavior that happened to us in the first place, with the query just hanging infinitely. We need to handle this validation error, either here on the receiver side, or on the senders' sides, and fail the query.
2. With that in mind, I think we should set our own custom limit based on common sense. Like, idk, 2 gigabytes.
3. This whole behavior is kind of stupid: I am running a query that produces 100 gigabytes, why can't the system give me the first 16MB? It should be able to truncate it. Let's think about implementing truncation in the engines themselves, at least in the future.